scripts/example_script.py

```diff
@@ -0,0 +1,241 @@
+gc = girder_client.GirderClient(apiUrl=girder_location)
+gc.authenticate(interactive=True)
+
+# get the id of the desired collection
+try:
+    collection_data = gc.getResource(request_url)
+except HttpError as err:
+    print_http_error(err)
+    exit()
+
+collection_id = collection_data['_id']
+new_folder_data = None
+# create a folder under this collection or load a preexisting folder that exists
+# with the same name and parent
+try:
+    new_folder_data = gc.load_or_create_folder(new_folder_name,
+                                               collection_id, 'collection')
+except HttpError as err:
+    print_http_error(err)
+    exit()
+new_folder_id = new_folder_data['_id']
+
+# get all folders in the collection
+collection_composition = None
+try:
+    collection_composition = gc.listFolder(collection_id, 'collection')
+except HttpError as err:
+    print_http_error(err)
+    print ('exited')
+    exit()
+
+# get the ids of all folders under the collection
+collection_folders = get_field(collection_composition, '_id')
+item_list = []
+# create a list to store id of all svs items
+img_list = []
+# get a list of all items within each folder of the collection
+recurse_get_resource(gc, collection_id, 'item', item_list,
+                     parent_is_collection=True)
+
+temp_img_id = None
+print('Listing all %s items in the collection %s' % (img_type, collection_name))
+for item in item_list:
+    if img_type in item['name']:
+        temp_img_id = item['_id']
+        print('image name %s ' % (item['name']))
+        img_list.append(temp_img_id)
+        turn_into_a_large_image(gc, temp_img_id)
+
+# now create PATIENT folders inside our new folder
+
+patient_folder_list = []
+patient_folder_name = 'Patient%d'
+
+for i in range(1, 4):
+    temp_name = patient_folder_name % (i)
+    try:
+        patient_folder_list.append(gc.load_or_create_folder(temp_name,
+                                                            new_folder_id,
+                                                            'folder'))
+    except HttpError as err:
+        print_http_error(err)
+        print ('exited')
+        exit()
+
+# now lets move an image into the PATIENT1 folder
+
+patient1_folder = patient_folder_list[0]
+patient1_folder_id = patient1_folder['_id']
+
+src_img_id = img_list[0]
+
+move_params = {
+    'resources': json.dumps({
+        'item': [str(src_img_id)]}),
+    'parentType': 'folder',
+    'parentId': patient1_folder_id
+}
+
+try:
+    gc.put('resource/move', move_params)
+except HttpError as err:
+    print_http_error(err)
+    print ('exited')
+    exit()
+
+# now lets edit the PATIENT1 folder
+
+patient1_meta_data = {'PatientAge': 22, 'PatientDeceased': True}
+
+try:
+    gc.addMetadataToFolder(patient1_folder_id, patient1_meta_data)
+except HttpError as err:
+    print_http_error(err)
+    print ('exited')
+    exit()
+
+# lets add meta data to the image we moved
+# note how the id remains the same
```

Comment:
Comments like this a great, noting specific features that may not be immediately obvious.